code/modules/holomap/holomap.dm

```diff
@@ -0,0 +1,660 @@
+/*
+	Somone should totally add an action button to select a location when observing the map
+	and it display a list to select from (TGUI maybe!? color coded !? YES?)
+	then show an icon like you_are_here there
+*/
+
+///this will hold images rendered at roundstart
+GLOBAL_LIST_EMPTY(holomaps)
+
+/datum/hud/holomap
+	inventory_shown = FALSE
+
+/datum/hud/holomap/New(list/r, zed, x_in, y_in, z_in)
+	..()
+	var/obj/screen/holomap/using
+	var/obj/screen/you_are_here/arrow
+	var/icon_key = "[zed]_[r[1]]-[r[2]]-[r[3]]-[r[4]]"
+
+	if(GLOB.holomaps[icon_key])
+		using = GLOB.holomaps[icon_key]
+	else
+		using = new/obj/screen/holomap(r, zed)
+		using.screen_loc = ui_holomap
+		GLOB.holomaps[icon_key] = using
+
+	static_inventory += using
+
+	if(zed == z_in)
+		arrow = new/obj/screen/you_are_here(x_in, y_in)
+		static_inventory += arrow
+
+	for(var/obj/screen/S in (static_inventory))
+		S.hud = src
+
+/obj/screen/holomap
+	var/image/holomap_projection
+	var/list/region_selection = list()
+	var/zlevel_rendered
+
+/obj/screen/holomap/New(list/region, rendered)
+	..()
+
+	screen_loc = ui_holomap
+	region_selection = region
+	zlevel_rendered = rendered
+
+	var/icon/base_icon
+
+	base_icon = make_map()
+	var/icon/alpha_mask = icon('icons/effects/480x480.dmi', "scanline") //Scanline effect.
+	base_icon.AddAlphaMask(alpha_mask)
+	holomap_projection = image(base_icon, loc = vis_contents, layer = ABOVE_HUD_PLANE)
+	holomap_projection.plane = ABOVE_HUD_PLANE
+	holomap_projection.filters = filter(type = "drop_shadow", size = 8, color = "#ffffffd4")
+
+	appearance = holomap_projection
+
+/obj/screen/you_are_here
+	name = "You are here!"
+	icon = 'icons/effects/holomap_icons.dmi'
+	icon_state = "you_are_here"
```

Comment:
![image](https://github.com/ParadiseSS13/Paradise/assets/25063394/e4292abc-8d27-4904-a2ea-9251bc86dd65)

This is why I will always frown upon mass find->replace for things such as `YouAreHere` --> `you_are_here`

Reply:
AHHH THIS IS WHY MY NEW CODE WASNT WORKING LMFAO